payjoin/src/uri.rs

```diff
@@ -75,6 +75,37 @@ pub trait UriExt<'a>: sealed::UriExt {
     fn check_pj_supported(self) -> Result<PjUri<'a>, bip21::Uri<'a>>;
 }
 
+impl<'a> bip21::SerializeParams for &'a Payjoin {
+    type Key = &'static str;
+    type Value = String;
+    type Iterator = std::vec::IntoIter<(Self::Key, Self::Value)>;
+
+    fn serialize_params(self) -> Self::Iterator {
+        match self {
+            Payjoin::Supported(params) => params.serialize_params(),
+            Payjoin::V2Only(params) => params.serialize_params(),
+            Payjoin::Unsupported => vec![].into_iter(),
+        }
+    }
+}
+
+impl<'a> bip21::SerializeParams for &'a PayjoinParams {
+    type Key = &'static str;
+    type Value = String;
+    type Iterator = std::vec::IntoIter<(Self::Key, Self::Value)>;
+
+    fn serialize_params(self) -> Self::Iterator {
+        vec![
+            ("pj", self.endpoint.as_str().to_string()),
+            ("pjos", if self.disable_output_substitution { "1" } else { "0" }.to_string()),
+            #[cfg(feature = "v2")]
+            // this unwrap is safe because we know that the config is required for v2
```

Comment:
I think a future fix would be to make the ohttp_config non optional in the struct

Reply:
Such a decision would mean our library does not support creating Bip21 payjoin URIs for v1 payjoin. I think we want to be able to support v1 for services like exchanges that can host servers and don't need to run the added complexity of HPKE/OHTTP by default. But there is some merit to OHTTP by default.